src/Xamarin.Android.Build.Tasks/Utilities/MetadataExtensions.cs

```diff
@@ -12,8 +12,23 @@ public static string GetCustomAttributeFullName (this MetadataReader reader, Cus
 		{
 			if (attribute.Constructor.Kind == HandleKind.MemberReference) {
 				var ctor = reader.GetMemberReference ((MemberReferenceHandle)attribute.Constructor);
-				var type = reader.GetTypeReference ((TypeReferenceHandle)ctor.Parent);
-				return reader.GetString (type.Namespace) + "." + reader.GetString (type.Name);
+				try {
+					if (ctor.Parent.Kind == HandleKind.TypeReference) {
+						var type = reader.GetTypeReference ((TypeReferenceHandle)ctor.Parent);
+						return reader.GetString (type.Namespace) + "." + reader.GetString (type.Name);
+					} else if (ctor.Parent.Kind == HandleKind.TypeSpecification) {
+						var type = reader.GetTypeSpecification ((TypeSpecificationHandle)ctor.Parent);
+						BlobReader blobReader = reader.GetBlobReader (type.Signature);
+						SignatureTypeCode typeCode = blobReader.ReadSignatureTypeCode ();
+						EntityHandle typeHandle = blobReader.ReadTypeHandle ();
+						TypeReference typeRef = reader.GetTypeReference ((TypeReferenceHandle)typeHandle);
+						return reader.GetString (typeRef.Namespace) + "." + reader.GetString (typeRef.Name);
+					}
```

Comment:
then here, should we have a block like:
```csharp
else
{
    LogDebugMessage($"Unknown EntityHandle.Kind: {ctor.Parent.Kind}");
    return null;
}
```

Reply:
lol "Add Logging" 👍  now we change 4 other files 🤣